src/components/App.js

```diff
@@ -1,8 +1,196 @@
+import React, { useState } from 'react';
+import SelectNumber from './Select';
+import Board from './Board';
+import { alphabet } from '../letters/Alphabet';
+import { wordDictionary2 } from '../letters/WordDictionary2';
+import { wordDictionary3 } from '../letters/WordDictionary3';
+import { wordDictionary4 } from '../letters/WordDictionary4';
+import { wordDictionary5 } from '../letters/WordDictionary5';
+import { letterPoints } from '../letters/LetterPoints';
 
 function App() {
+  const [selectNumber, setSelectNumber] = useState(false);
+  const [wordNo, setWordNo] = useState(1);
+  const [numberSelected, setNumberSelected] = useState('');
+  const [squares, setSquares] = useState([]);
+  const [wordLengths, setWordLengths] = useState([]);
+  const [remainingAlphabet, setRemainingAlphabet] = useState(alphabet);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [score, setScore] = useState(0);
+  let game;
+  let words = [];
+  let invalidWord = '';
+
+  class Game {
+    restart() {
+      setSelectNumber(false);
+      setNumberSelected('');
+      setWordLengths([]);
+      setSquares([]);
+      setWordNo(1);
+      setErrorMessage('');
+      setRemainingAlphabet(alphabet);
+      setScore(0);
+      console.clear();
+    }
+    calculateScore(words) {
+      let workScore = 20 - numberSelected;
+      for (let i = 0; i < words.length; i++) {
+        for (let j = 0; j < words[i].length; j++) {
+          workScore =
+            workScore +
+            letterPoints.find(item => {
+              return item.letter === words[i][j].toUpperCase();
+            }).point;
+      }
+    }
+    return workScore;
+    }
+  };
+  game = new Game();
+
+  const handleClick = () => {
+    // which squares contain 1's,2's,3's,.....
+    let wordN;
```

Comment:
most of this should probably be put into a helper called like, getWords

Reply:
In general you should try to take your basic logic and turn them into helper functions that simply take input and return output, with no side effects (side effects being like setting state)

Then your `handleClick` uses the helpers to get results and does the side effects

Reply:
Done. However I'm not sure I can pass parms which get updated in a called util function.

Luke.....can you comment please?

Reply:
Never mind......I fixed this..